internal/fs/tarfs/tarfs.go

```diff
@@ -135,8 +136,16 @@ func (tfs *TarFS) indexEntries() error {
 		header, err := tr.Next()
 		if err != nil {
 			if errors.Is(err, io.EOF) {
+				if len(tfs.entries) == 0 {
+					// indicates the given file is empty
+					return errdef.ErrNotTarFile
+				}
 				break
 			}
+			if errors.Is(err, io.ErrUnexpectedEOF) {
+				// indicates that its either not a tarfile or it is a corrupted one
+				return errdef.ErrNotTarFile
```

Comment:
We can also wrap this error with a friendly error message like "{path} is not a valid tar file"

Reply:
Wrapped the error with the path of the file.